src/applications/gi/reducers/eligibility.js

```diff
@@ -22,27 +22,39 @@ export default function(state = INITIAL_STATE, action) {
   if (action.type === ELIGIBILITY_CHANGED) {
     const { field, value } = action;
 
-    const newState = {
+    let newState = {
       ...state,
       [field]: value,
     };
 
     if (field === 'militaryStatus') {
-      return set('spouseActiveDuty', 'no', newState);
+      newState = {
+        ...newState,
+        spouseActiveDuty: 'no',
+      };
     }
 
     if (field === 'giBillChapter') {
-      return merge(newState, {
+      newState = {
+        ...newState,
         cumulativeService: '1.0',
         enlistmentService: '3',
         consecutiveService: '0.8',
         eligForPostGiBill: 'no',
         numberOfDependents: '0',
-      });
+      };
+    }
+
+    // Test fix for 7528 and 8228
+    if (!environment.isProduction()) {
+      localStorage.setItem('giEligibility', JSON.stringify(newState));
     }
 
     return newState;
   }
 
-  return state;
+  return {
+    ...state,
+    ...JSON.parse(localStorage.getItem('giEligibility')),
+  };
```

Comment:
Do we need the prod flag here also?

Reply:
If it's never set, it's just `undefined` and won't cause an issue